src/components/App/App.test.js

```diff
@@ -1,9 +1,88 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './App';
-
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
-});
+import React from 'react'
+import App from './App'
+import characterResults from '../../mockData/fakeCharacters.js'
+import { shallow } from 'enzyme'
+import { BrowserRouter } from 'react-router-dom'
+import { getMoviesData } from '../../apiCalls/apiCalls'
+
+jest.mock('../../apiCalls/apiCalls')
+
+describe("App", () => {
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = shallow(< App />, { wrapper: BrowserRouter })
+  })
+
+  it('should render App', () => {
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  it('should call function to fetch api data', () => {
+    getMoviesData.mockImplementation(() => {
```

Comment:
So my understanding here is that you called jest.mock on apiCalls on line 8 so that you can run some incantations to mock out the async calls, such as .mockImplementation() on line 32.  Just checking my understanding here!

Reply:
this is keeping the .thens from crashing the test?